improver_tests/calibration/reliability_calibration/conftest.py

```diff
@@ -0,0 +1,308 @@
+    var_name="threshold",
+    units="K",
+    attributes={"spp__relative_to_threshold": "above"},
+)
+
+
+@pytest.fixture
+def expected_table_shape_grid():
+    return 3, 5, 3, 3
+
+
+@pytest.fixture
+def expected_table_shape_spot():
+    return 3, 5, 9
+
+
+@pytest.fixture
+def expected_attributes():
+    return {
+        "title": "Reliability calibration data table",
+        "source": "IMPROVER",
+        "institution": "unknown",
+    }
+
+
+@pytest.fixture
+def forecast_grid():
+    forecast_data = np.arange(9, dtype=np.float32).reshape(3, 3) / 8.0
+    forecast_data_stack = np.stack([forecast_data, forecast_data])
+    forecast_1 = set_up_probability_cube(forecast_data_stack, _thresholds)
+    forecast_2 = set_up_probability_cube(
+        forecast_data_stack,
+        _thresholds,
+        time=datetime(2017, 11, 11, 4, 0),
+        frt=datetime(2017, 11, 11, 0, 0),
+    )
+    forecasts_grid = MergeCubes()([forecast_1, forecast_2])
+    return forecasts_grid
+
+
+@pytest.fixture
+def truth_grid():
+    truth_data = np.linspace(281, 285, 9, dtype=np.float32).reshape(3, 3)
+    # Threshold the truths, giving fields of zeroes and ones.
+    truth_data_a = (truth_data > _thresholds[0]).astype(int)
+    truth_data_b = (truth_data > _thresholds[1]).astype(int)
+    truth_data_stack = np.stack([truth_data_a, truth_data_b])
+    truth_1 = set_up_probability_cube(
+        truth_data_stack, _thresholds, frt=datetime(2017, 11, 10, 4, 0)
+    )
+    truth_2 = set_up_probability_cube(
+        truth_data_stack,
+        _thresholds,
+        time=datetime(2017, 11, 11, 4, 0),
+        frt=datetime(2017, 11, 11, 4, 0),
+    )
+    truths_grid = MergeCubes()([truth_1, truth_2])
+    return truths_grid
+
+
+@pytest.fixture
+def forecast_spot(forecast_grid):
+    forecast_data = forecast_grid.data[0, ...]
+    spot_probabilities = forecast_data.reshape((2, 9))
+    forecasts_spot_list = CubeList()
+    for day in range(5, 7):
+        time_coords = construct_scalar_time_coords(
+            datetime(2017, 11, day, 4, 0), None, datetime(2017, 11, day, 0, 0),
+        )
+        time_coords = [t[0] for t in time_coords]
+        forecasts_spot_list.append(
+            build_spotdata_cube(
+                spot_probabilities,
+                name="probability_of_air_temperature_above_threshold",
+                units="1",
+                altitude=_dummy_point_locations,
+                latitude=_dummy_point_locations,
+                longitude=_dummy_point_locations,
+                wmo_id=_dummy_string_ids,
+                additional_dims=[_threshold_coord],
+                scalar_coords=time_coords,
+            )
+        )
+    forecasts_spot = forecasts_spot_list.merge_cube()
+    return forecasts_spot
+
+
+@pytest.fixture
+def truth_spot(truth_grid):
+    print(truth_grid)
+    print(truth_grid.data.shape)
+    truth_data_spot = truth_grid[0, ...].data.reshape((2, 9))
+    print(truth_data_spot.shape)
+    print(truth_data_spot)
```

Comment:
Almost forgot, testing cruft

Reply:
These (and two others) are deleted in 5756846.